src/libaktualizr/primary/reportqueue.h

```diff
@@ -78,8 +79,7 @@ class ReportQueue {
   std::condition_variable cv_;
   std::mutex thread_mutex_;
   std::mutex queue_mutex_;
-  std::queue<std::unique_ptr<ReportEvent>> report_queue_;
-  Json::Value report_array{Json::arrayValue};
+  std::vector<std::pair<std::shared_ptr<std::promise<bool>>, std::unique_ptr<ReportEvent>>> report_queue_;
```

Comment:
Would it be realistic to embed the promise inside `ReportEvent`? It could simplify this class a bit.

Reply:
Yes I just embeded, but not sure is it better.

Reply:
If you don't think it's a good change, we can talk about it that's fine.

It's just that when `enqueue` is called, the ownership of `event` is passed to the queue anyway so it should be able to modify it and use an internal member for bookkeeping.